### PR TITLE
Zack/inventory

### PR DESCRIPTION
**Pull request by Zhirschhorn**
I created two new screens:
-InventoryModificationScreen: This basically just takes the InventoryView component and adds an Add Item button at the bottom. Related to this, I also changed the InventoryView to take a new prop called collapsible. When set to false, the inventory view will start full screen and will no be able to be swiped down. It is an optional prop that defaults to true so it will not break previous code.

-Add Item Screen: This is where I spent most of my time. I used expo Camera (https://docs.expo.io/versions/latest/sdk/camera/) and expo ImagePicker (https://docs.expo.io/versions/latest/sdk/imagepicker/) to implement the images. When a picture is taken/chosen, the camera is replaced with that image. I used React Native Elements' Input for the inputs underneath. It checks the cost and quantity inputs and will show an error/not submit if they are not proper input. 

This is my first time writing react code so I'm sure there will be a lot of issues, lmk and I'll be happy to fix them

Notion: https://www.notion.so/BruinBot-Inventory-Modification-Forms-c197d855843e421b867bde043a1eef9a

AddItemWithPicture
AddItemCamera
InventoryModification

### CODE REVIEW

**Comment by zhang-edward** [comment]
I messed up the GitHub actions so it didn't catch this, but you should run Prettier to fix formatting errors! `npm lint:fix`

File: src/containers/AddItemScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            const pic = await camera.takePictureAsync({skipProcessing:true});
+            setPhoto(pic.uri);
+        }
+    };
+
+    const pickImage = async () => {
+        const pic = await ImagePicker.launchImageLibraryAsync({
+          mediaTypes: ImagePicker.MediaTypeOptions.Images,
+        });
+        if (pic.cancelled){
```
Comment (zhang-edward) [inline src/containers/AddItemScreen.tsx:45]: nitpick
```suggestion
     setPhoto(pic.cancelled ? null : pic.uri)
```

File: src/containers/AddItemScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+          setHasPermission(response.status === 'granted');
+        })();
+    }, []);
+
+    if (hasPermission === null) {
+        return (
+			<Text> Requesting permission </Text>
+        );      
+    }
+    if (hasPermission === false) {
```
Comment (zhang-edward) [inline src/containers/AddItemScreen.tsx:63]: nitpick
```suggestion
    else if (hasPermission === false) {
```

File: src/containers/InventoryModification.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import { TouchableOpacity, Dimensions, StyleSheet, View, Text, Alert } from 'react-native';
+import { ItemProps, InventoryProps } from "../types/inventoryTypes"
+import Ham from '../assets/greenHam.jpg';
+import Crane from '../assets/crane.png';
+import Inventory from "../components/InventoryView";
+
+const screenWidth = Dimensions.get('window').width;
+
+interface InventoryModificationProps {
+    navigation: any;
```
Comment (zhang-edward) [inline src/containers/InventoryModification.tsx:11]: Again, don't use `any` here

File: src/containers/InventoryModification.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    botItems["123"] = item;
+    
+    const [infor, setInfor] = useState(botInfo);
+	const [items, setItems] = useState(botItems);
+
+	return (
+        <>
+		<View style={{ flex:1}}>
+			<Inventory id="123" info={infor} items={items} collapsable={false} />
+			<TouchableOpacity style={styles.button} onPress={() => {
```
Comment (zhang-edward) [inline src/containers/InventoryModification.tsx:78]: Could we use the React Native Elements button here instead?
https://reactnativeelements.com/docs/button/

File: src/containers/AddItemScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+                }}/>
+        <Text style = {styles.header}> Add an Item </Text>
+        <View style = {styles.cameraContainer}>
+            { (photo === null) ? (
+            <>
+            <Camera style={styles.camera} type={type} 
+            ref={ref =>{
+                setCamera(ref);
+            }}/>
+            <Ionicons name="ios-photos" style={{
```
Comment (zhang-edward) [inline src/containers/AddItemScreen.tsx:101]: I would recommend using icons from React Native Elements and choosing a consistent icon set (I think we're just sticking to FontAwesome for now)

https://reactnativeelements.com/docs/icon
Icon name: https://fontawesome.com/icons?d=gallery

File: src/containers/AddItemScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+                color: "#fff", 
+                fontSize: 50,
+                backgroundColor: 'transparent',
+                position: 'absolute',
+                bottom: 5,
+                left: screenWidth * .11
+            }} onPress={()=>{
+                pickImage();
+            }}/>
+            <FontAwesome name="circle" style={{
```
Comment (zhang-edward) [inline src/containers/AddItemScreen.tsx:111]: See above comment

File: src/containers/AddItemScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+            }} onPress={()=>{
+                takePhoto();
+            }}/>
+            </>
+            ) : ( 
+            <>
+                <Image style={styles.camera} source={{uri: photo}}
+                PlaceholderContent={<ActivityIndicator />}
+                ></Image>
+                <FontAwesome name="times" style={
```
Comment (zhang-edward) [inline src/containers/AddItemScreen.tsx:126]: Same thing here

File: src/containers/InventoryModification.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    botInfo["123"] = {
+        name: "bot 1",
+        inventorySize: 10,
+        distance: 0,
+        itemsSold: 0,
+        imgSrc: Crane
+    };
+    botItems["123"] = item;
+    
+    const [infor, setInfor] = useState(botInfo);
```
Comment (zhang-edward) [inline src/containers/InventoryModification.tsx:71]: Nitpick but could we call this `info` instead?

Also, introducing an underscore before variable names will fix the warning about unused vars, if we aren't using for now but are planning to in the future.

```suggestion
    const [info, _setInfo] = useState(botInfo);
```

**Review by zhang-edward — CHANGES_REQUESTED** [review]
See above comment about running `npm lint:fix` and fixing those errors!

**Review by zhang-edward — APPROVED** [review]

**Comment by zhang-edward** [comment]
Let's fix these merge conflicts too. 

For the package files, just pull main and then just regenerate the `package-lock.json` and `package.json` files!

**Comment by Zhirschhorn** [comment]
> Let's fix these merge conflicts too.
> 
> For the package files, just pull main and then just regenerate the `package-lock.json` and `package.json` files!

How do I regenerate the 'package-lock.json' and 'package.json' files?

**Comment by Zhirschhorn** [comment]
> Let's fix these merge conflicts too.
> 
> For the package files, just pull main and then just regenerate the `package-lock.json` and `package.json` files!

**Comment by Zhirschhorn** [comment]
Also I accidently

**Comment by dvorjackz** [comment]
> > Let's fix these merge conflicts too.
> > For the package files, just pull main and then just regenerate the `package-lock.json` and `package.json` files!
> 
> How do I regenerate the 'package-lock.json' and 'package.json' files?

1. Fix the merge conflict in package.json and app.tsx. Basically go to main, pull in the changes, then `git merge main` on your branch. Looks like to fix the merge conflicts, you just need to include both your changes and the incoming changes from main.
2. Delete package-lock.json and `npm i`
3. Complete the merge by adding all three files and committing

**Pull request merged**